### PR TITLE
Remove unused MUQ_HAS_HDF5

### PR DESCRIPTION
**Pull request by pierrepebay**

### CODE REVIEW

File: cmake/finds/RequiredDependencySearch.cmake [hunk 1]
```diff
@@ -68,7 +68,6 @@ GetDependency(OTF2)
 set(HAVE_HDF5 1)
```
Comment (fnrizzi) [inline cmake/finds/RequiredDependencySearch.cmake:68]: is HAVE_HDF5 needed?

Comment (pierrepebay) [reply cmake/finds/RequiredDependencySearch.cmake:68]: I was actually just checking this, same thing for NLopt, they don't appear to be used.

Comment (fnrizzi) [reply cmake/finds/RequiredDependencySearch.cmake:68]: please remove it

Comment (pierrepebay) [reply cmake/finds/RequiredDependencySearch.cmake:68]: Removed in https://github.com/NexGenAnalytics/MIT-MUQ/pull/32

Comment (fnrizzi) [reply cmake/finds/RequiredDependencySearch.cmake:68]: i mean remove in this PR the HAVE_HDF5

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**